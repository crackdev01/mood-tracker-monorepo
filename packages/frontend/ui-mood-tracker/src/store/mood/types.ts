// Define Mood Store State.
export interface MoodState {
  mood: [];
}

// Define Mood Store actions.
export enum MoodActions {
  FETCH_MOODS = 'FETCH_MOODS',
  RENDER_MOODS = 'RENDER_MOODS',
}
