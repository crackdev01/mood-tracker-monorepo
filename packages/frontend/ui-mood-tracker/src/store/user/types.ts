export interface UserRequest {
  username: string;
  password: string;
}

export interface Jwt {
  exp: number;
  iat: number;
  username: string;
  uuid: string;
}

export interface User {
  accessToken: string;
  decodedAccessToken: Jwt | null;
}

// Define User Store State.
export interface UserState {
  user: User;
}

// Define User Store actions.
export enum UserActions {
  FETCH_USER = 'FETCH_USER',
  RENDER_USER = 'RENDER_USER',
  LOGOUT_USER = 'LOGOUT_USER',
}
