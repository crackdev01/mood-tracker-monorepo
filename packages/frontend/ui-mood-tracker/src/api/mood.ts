import { authMoodTrackerApi } from './core';

// FIXME: update types.
export async function getMoodEntries(): Promise<any> {
  return await authMoodTrackerApi.get('/mood/entries');
}

export async function postMoodEntry(payload: any): Promise<void> {
  return await authMoodTrackerApi.post('/mood/entry', payload);
}

export async function updateMoodEntry(payload: any): Promise<void> {
  return await authMoodTrackerApi.put('/mood/entry', payload);
}

export async function deleteMoodEntry(payload: any): Promise<void> {
  return await authMoodTrackerApi.delete(`/mood/entry/${payload.id}`);
}
