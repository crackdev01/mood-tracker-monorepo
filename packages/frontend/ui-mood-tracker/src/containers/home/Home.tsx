import React from 'react';
import { useSelector } from 'react-redux';
import { BrowserRouter, Route, Switch } from 'react-router-dom';

import ProtectedRoute from '../../components/ProtectedRoute';
import Login from '../login/Login';
import MoodEntry from '../mood-entry/MoodEntry';
import Statistics from '../statistics/Statistics';
import SiteHeader from '../../components/site-header/SiteHeader';
import { ApplicationState } from '../../store';

import './home.scss';

const Home = () => {
  const user = useSelector((state: ApplicationState) => state.userReducer.user);

  const isAuthenticated = !!user.accessToken;

  return (
    <BrowserRouter>
      <SiteHeader />

      <article className="app-content">
        <Switch>
          <Route exact path="/">
            {isAuthenticated && <MoodEntry />}
            {!isAuthenticated && <Login />}
          </Route>
          <ProtectedRoute
            path="/mood-entry"
            isAuthenticated={isAuthenticated}
            component={MoodEntry}
          />
          <ProtectedRoute
            path="/statistics"
            isAuthenticated={isAuthenticated}
            component={Statistics}
          />
        </Switch>
      </article>
    </BrowserRouter>
  );
};

export default Home;
