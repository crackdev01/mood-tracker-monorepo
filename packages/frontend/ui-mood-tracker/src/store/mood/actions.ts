import { MoodActions } from './types';

export const fetchMoodEntries = () => {
  return {
    type: MoodActions.FETCH_MOODS,
  };
};

export const addMoodEntry = () => {
  return {
    type: MoodActions.ADD_MOOD,
  };
};

export const editMoodEntry = () => {
  return {
    type: MoodActions.EDIT_MOOD,
  };
};

export const deleteMoodEntry = () => {
  return {
    type: MoodActions.DELETE_MOOD,
  };
};

// FIXME: fix type.
export const renderMoodEntries = (mood: any) => {
  return {
    type: MoodActions.RENDER_MOODS,
    mood,
  };
};
