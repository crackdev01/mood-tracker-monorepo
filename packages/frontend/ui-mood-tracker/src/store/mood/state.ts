import { MoodState } from './types';

export const moodState: MoodState = {
  mood: [],
};
