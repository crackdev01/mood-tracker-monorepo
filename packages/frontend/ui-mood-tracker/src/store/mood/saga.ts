import { all, call, put, takeEvery, fork } from 'redux-saga/effects';

import { MoodActions } from './types';
import { getMoodEntriesApi, postMoodEntry, updateMoodEntry, deleteMoodEntry } from '../../api/mood';

// FIXME: update types in file.

// Fetch mood entries.
export function* getMoodEntriesCall() {
  const { data } = yield call(getMoodEntriesApi);
  yield put({ type: MoodActions.RENDER_MOODS, data });
}

export function* getMoodEntries() {
  yield takeEvery(MoodActions.FETCH_MOODS, getMoodEntriesCall);
}

// Add mood entry.
export function* addMoodEntryCall(action: any) {
  yield call(postMoodEntry, action.payload);
  yield call(getMoodEntriesCall);
}

export function* addMoodEntry() {
  yield takeEvery(MoodActions.ADD_MOOD, addMoodEntryCall);
}

// Update mood entry.
export function* editMoodEntryCall(action: any) {
  yield call(updateMoodEntry, action.payload);
  yield call(getMoodEntries);
}

export function* editMoodEntry() {
  yield takeEvery(MoodActions.EDIT_MOOD, editMoodEntryCall);
}

// Delete mood entry.
export function* removeMoodEntryCall(action: any) {
  yield call(deleteMoodEntry, action.payload);
  yield call(getMoodEntries);
}

export function* removeMoodEntry() {
  yield takeEvery(MoodActions.DELETE_MOOD, removeMoodEntryCall);
}

export function* moodSaga() {
  yield all([fork(getMoodEntries), fork(addMoodEntry), fork(editMoodEntry), fork(removeMoodEntry)]);
}

export default moodSaga;
