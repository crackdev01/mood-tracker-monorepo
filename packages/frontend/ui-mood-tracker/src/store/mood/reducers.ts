import { moodState } from './state';
import { MoodActions } from './types';

export const mood = (state = moodState, action: any) => {
  switch (action.type) {
    case MoodActions.FETCH_MOODS:
      return {
        ...state,
        mood: action.mood,
      };
    default:
      return state;
  }
};
