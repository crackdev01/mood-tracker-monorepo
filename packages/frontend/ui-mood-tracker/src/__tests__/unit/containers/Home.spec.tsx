import React from 'react';
import { useSelector } from 'react-redux';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Home from '../../../containers/home/Home';
import { mockUser } from '../../fixtures';
import { MoodActions } from '../../../store/mood/types';

Enzyme.configure({ adapter: new Adapter() });

const useSelectorMock = useSelector as jest.Mock<any>;
const mockDispatch = jest.fn();

jest.mock('react-redux', () => ({
  useDispatch: () => mockDispatch,
  useSelector: jest.fn(),
}));
/* Enzyme doesn't support automatic triggering of useEffect on load yet.
 * So `useEffect` has to be mocked while requiring rest of `react`.
 * https://github.com/enzymejs/enzyme/issues/2086
 */
jest.mock('react', () => ({
  ...jest.requireActual('react') as any,
  useEffect: (f: any) => f(),
}));

describe('Home', () => {
  test('renders', () => {
    mockDispatch.mockReturnValueOnce({})
    useSelectorMock.mockImplementationOnce((s) => s({ userReducer: { user: mockUser } }));
    const w = shallow(<Home />);
    expect(w.debug()).toMatchSnapshot();
    expect(mockDispatch).toHaveBeenCalledWith({
      type: MoodActions.FETCH_MOODS,
    })
  });
});
