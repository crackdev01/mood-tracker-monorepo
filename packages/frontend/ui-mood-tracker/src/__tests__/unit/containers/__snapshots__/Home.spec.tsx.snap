// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders 1`] = `
"<BrowserRouter>
  <SiteHeader />
  <LocationModal />
  <article className=\\"app-content\\">
    <Switch>
      <Route exact={true} path=\\"/\\">
        <MoodEntry />
      </Route>
      <ProtectedRoute path=\\"/mood-entry\\" isAuthenticated={true} component={[Function: MoodEntry]} />
      <ProtectedRoute path=\\"/statistics\\" isAuthenticated={true} component={[Function: Statistics]} />
    </Switch>
  </article>
</BrowserRouter>"
`;
